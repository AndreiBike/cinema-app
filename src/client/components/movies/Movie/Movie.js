import React from 'react';
import "./Movie.module.css";

const Movie = (props) => {

  const {
    movie: {
      id,
      imageWay,
      name,
      year,
      gengre,
    }
  } = props;

  return (
    <div className="movie">
      <div className="movie-poster">
        <img src={imageWay} />
      </div>

      <div className="movie-filmname">
        <div>{name}</div>
        <div> {year} </div>
      </div>

      <div className="movie-gengre">
        {gengre}
      </div>
    </div>
  )
}

export default Movie;
