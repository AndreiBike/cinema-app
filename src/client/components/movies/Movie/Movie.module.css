.movie {
  height: 539px;
  width: 324px;
  margin-top: 36px;
}

.movie-poster {
  height: 455px;
  width: 324px;
  background-color: silver;
}

.movie-poster img{
  height: 100%;
  width: 100%;
}

.movie-filmname{
  margin-top: 22px;
  display: flex;
  justify-content: space-between;
}

.movie-filmname div:first-child {
  height: 22px;
  width: 232px;
  opacity: 0.7;
  color: white;
  font-family: Montserrat;
  font-size: 18px;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 22px;
}

.movie-filmname div:last-child {
  color: white;
  border: 1px solid #979797;
  width: 66px;
  height: 26px;
  opacity: 0.5;
  border-radius: 4px;
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  font-size: 14px;
  font-weight: 500;
  letter-spacing: 0;
}

.movie-gengre {
  margin-top: 8px;
  height: 18px;
  width: 323px;
  opacity: 0.5;
  color: white;
  font-family: Montserrat;
  font-size: 14px;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 18px;
}