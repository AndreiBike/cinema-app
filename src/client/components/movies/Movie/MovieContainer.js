import React from 'react';
import Movie from '@root/client/components/movies/movie/Movie';

class MovieContainer extends React.Component {

  constructor(props){
    super(props);
    this.state = {
      id: props.movie.id,
      imageWay: props.movie.imageWay,
      name: props.movie.name,
      year: props.movie.year,
      gengre: props.movie.gengre,
    }
  }

/*
  componentDidMount(){
    this.setState({
      id: this.props.movie.id,
      imageWay: this.props.movie.imageWay,
      name: this.props.movie.name,
      year: this.props.movie.year,
      gengre: this.props.movie.gengre,
    })
  }
*/
  render() {
    return (
      <>
        <Movie id = {this.state.id}
               imageWay = {this.state.imageWay}
               name = {this.state.name}
               year = {this.state.year}
               gengre = {this.state.gengre}
        />
      </>
    )
  }
}

export default MovieContainer