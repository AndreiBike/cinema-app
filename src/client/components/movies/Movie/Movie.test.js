import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Movie from './Movie'

configure({adapter: new Adapter()});

const movieTest = {
  id: 1,
  imageWay: './assets/poster.jpg',
  name: "Avengers",
  year: 2020,
  gengre: "Action film"
};

describe("<Movie> Snapshot", () => {
  const component = shallow(<Movie key = {movieTest.id} movie = {movieTest} />);
  it('+++capturing snapshot of Movie',()=>{
    expect(component).toMatchSnapshot();
  });
});