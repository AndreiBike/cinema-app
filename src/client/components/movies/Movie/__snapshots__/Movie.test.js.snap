// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Movie> Snapshot +++capturing snapshot of Movie 1`] = `ShallowWrapper {}`;
